### PR TITLE
Menu component

### PR DESCRIPTION
**Pull request by tomaskikutis**
SDCP-545

### CODE REVIEW

File: examples/pages/react/Menu.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            <section className='docs-page__container'>
+
+                <h2 className='docs-page__h2'>Menu</h2>
+
+                <p className='docs-page__paragraph'></p>
+                <Markup.ReactMarkup>
+                    <Markup.ReactMarkupPreview>
+                        <div className='docs-page__content-row docs-page__content-row--no-margin'>
+                            <div className='form__row'>
+                                <input type="text" />
```
Comment (pavlovicnemanja) [inline examples/pages/react/Menu.tsx:73]: I think this is not needed.

File: examples/pages/react/Menu.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        label: 'Spike item',
+        icon: 'icon-trash',
+        onClick: () => console.log('action: Spike item'),
+    },
+];
+
+export class MenuDocs extends React.Component {
+    render() {
+        return (
+            <section className='docs-page__container'>
```
Comment (pavlovicnemanja) [inline examples/pages/react/Menu.tsx:64]: Can you add minimal example code at top like on other pages?

image

Also include small description and props list?

File: package.json [hunk 3]
```diff
@@ -82,12 +82,12 @@
   "dependencies": {
     "@material-ui/lab": "^4.0.0-alpha.56",
     "@popperjs/core": "^2.4.0",
+    "@superdesk/primereact": "^5.0.2-3",
```
Comment (pavlovicnemanja) [inline package.json:85]: Is it safe to go to version 6x?

File: examples/pages/react/Menu.tsx (hunk 1 above)
Comment (tomaskikutis) [reply examples/pages/react/Menu.tsx]: I'll remove it. I initially had it in order to test the focus easily.

File: examples/pages/react/Menu.tsx (hunk 2 above)
Comment (tomaskikutis) [reply examples/pages/react/Menu.tsx:64]: I have added example code in the markup section.

Regarding props, there is an interface defined that specifies all possible configurations. Why rewrite the same info in the docs? Whoever is going to use a component can jump to the interface in a single click. I'm happy to add comments to interface properties if you find some unclear.

go-to-interface

File: package.json (hunk 3 above)
Comment (tomaskikutis) [reply package.json:85]: I don't think so. It's a major release so we would need to re-test all components that we use.

File: examples/pages/react/Menu.tsx (hunk 2 above)
Comment (pavlovicnemanja) [reply examples/pages/react/Menu.tsx:64]: I agree, but we also use the props list so developers can easily see if component covers everything they need. All major frameworks use this approach so we want to have it also. All over components have it.

Comment (tomaskikutis) [reply examples/pages/react/Menu.tsx:64]: Doesn't jumping to the interface allow to easily see the same info? :smile:

Regarding other frameworks, we can borrow good practices that are useful for us, but we don't need to copy everything. For example, a lot of frameworks are written in js and don't have interfaces, so documenting props makes sense for them. We do have interfaces, so we aren't getting much value from describing the same info in a different way. It's more of a maintenance burden for us.

@petrjasek what do you think?

Comment (pavlovicnemanja) [reply examples/pages/react/Menu.tsx:64]: In that case you need to have imported that component to see if there is functionality you need. If you have everything listed, you can just go through components and see which one suits you the best. This is how we work, so please let's keep it consistant.

Comment (fritzSF) [reply examples/pages/react/Menu.tsx:64]: Come on @tomaskikutis, please add the damn list. I don't see what the big deal is, it won't take much of your time. It'll probably take less effort than going into a discussion about it. I get that you or other developers might not need it but this is not just for developers. Non-developers, I included, can use the components to create more complex layouts, playground pages etc. It's good to have an overview without having to dig into the code. Please, let's keep the practice we started.

**Review by pavlovicnemanja — APPROVED** [review]
LGTM! Just fix conflicts and merge

**Pull request merged**